src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -205,11 +205,32 @@
     @GET
     @Path("/{id}/history")
     @Produces(APPLICATION_JSON)
+    @Parameters({
+            @Parameter(
+                    name = "pageSize",
+                    in = ParameterIn.QUERY,
+                    description = "Number of items per page, if not specified or 0 is used, returns all elements",
+                    schema = @Schema(type = SchemaType.INTEGER)
+            ),
+            @Parameter(
+                    name = "pageNumber",
+                    in = ParameterIn.QUERY,
+                    description = "Page number. Starts at first page (0), if not specified starts at first page.",
+                    schema = @Schema(type = SchemaType.INTEGER)
+            ),
+            @Parameter(
+                    name = "include_detail",
+                    description = "Include the detail in the reply",
+                    schema = @Schema(type = SchemaType.BOOLEAN)
+            )
+    })
+
     @RolesAllowed(RbacIdentityProvider.RBAC_READ_INTEGRATIONS_ENDPOINTS)
-    public Uni<List<NotificationHistory>> getEndpointHistory(@Context SecurityContext sec, @PathParam("id") UUID id) {
+    public Uni<List<NotificationHistory>> getEndpointHistory(@Context SecurityContext sec, @PathParam("id") UUID id, @QueryParam("include_detail") Boolean includeDetail, @BeanParam Query query) {
```

Comment:
Query params are usually camel-cased, `include_detail` should be replaced with `includeDetail` for consistency.